apollo-router/src/plugins/authentication/mod.rs

```diff
@@ -167,7 +171,17 @@ fn search_jwks(
 ) -> Result<Option<(Option<String>, Jwk)>, BoxError> {
     const HIGHEST_SCORE: usize = 2;
     let mut candidates = vec![];
-    for JwkSetInfo { jwks, issuer } in jwks_manager.iter_jwks() {
+    for JwkSetInfo {
+        jwks,
+        issuer,
+        algorithms,
+    } in jwks_manager.iter_jwks()
+    {
+        // filter accepted algorithms
```

Comment:
NIT: Do we really need the is_empty() check first? If algorithms is a `HashSet` I would expect contains to be fast...

Reply:
Is empty is the case where no filtering is set. Would you prefer an Option there?

Reply:
moved to an option in [965f4ad](https://github.com/apollographql/router/pull/2852/commits/965f4adf84dedb6f014bc241b2ebcb1ce04ebafd)